source/common/upstream/cluster_manager_impl.h

```diff
@@ -280,10 +280,9 @@ class ClusterManagerImpl : public ClusterManager,
     return clusters_maps;
   }
 
-  OptRef<const Cluster> getActiveCluster(absl::string_view cluster_name) const override {
+  OptRef<const Cluster> getActiveCluster(const std::string& cluster_name) const override {
     ASSERT_IS_MAIN_OR_TEST_THREAD();
-    if (const auto& it = active_clusters_.find(std::string(cluster_name));
-        it != active_clusters_.end()) {
+    if (const auto& it = active_clusters_.find(cluster_name); it != active_clusters_.end()) {
```

Comment:
We actually could change the ClusterMap to use absl::btree_map and then use absl::string_view directly rather than change interface.

Reply:
Although I personally think the btree_map should works well, esp the map is only used for configuration loading. But I think you worry is also reasonable and change it to const string ref is much safer.

Reply:
I thought that `absl::flat_hash_map` supported heterogenous lookup.

> https://abseil.io/docs/cpp/guides/container
> 
> Hash Tables
> The Abseil container library contains a number of useful hash tables generally adhering to the STL container API contract:
> 
> absl::flat_hash_map
>  
>  ...
>  
> Supports heterogeneous lookup.

Does that not work in this case?

Reply:
We need a sorted map here for stable output of configdump. So, basically std::map or absl::btree_map.

Reply:
Sorry, I think I'm missing something. I think that the type of `active_clusters_` here is `ClusterInfoMap` defined to be:

`using ClusterInfoMap = absl::flat_hash_map<std::string, std::reference_wrapper<const Cluster>>;`

And I *think* `absl::flat_hash_map` supports heterogenous lookup so `absl::string_view` should just work here. But presumably some part of my thinking here is wrong?

Reply:
It's ClusterMap, not the cluster info map.
```
  // This map is ordered so that config dumping is consistent.
  using ClusterMap = std::map<std::string, ClusterDataPtr>;

  ClusterMap active_clusters_;
```

Reply:
Thanks @wbpcode for the offline conversation. LGTM!